pkg/services/object_manager/tombstone/getter.go

```diff
@@ -0,0 +1,94 @@
+package tombstone
+
+import (
+	"context"
+	"strconv"
+
+	lru "github.com/hashicorp/golang-lru"
+	objectV2 "github.com/nspcc-dev/neofs-api-go/v2/object"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	addressSDK "github.com/nspcc-dev/neofs-sdk-go/object/address"
+	"go.uber.org/zap"
+)
+
+// Source is a tombstone source interface.
+type Source interface {
+	// Tombstone must return tombstone from the source it was
+	// configured to fetch from and any error that appeared during
+	// fetching process.
+	//
+	// Note: it MUST NOT return any "not found"/"deleted" errors.
+	// Missing tombstone must lead to returning nil tombstone pointer
+	// and nil error.
+	Tombstone(ctx context.Context, a *addressSDK.Address, epoch uint64) (*object.Object, error)
+}
+
+// Getter is a tombstone source wrapper.
+// It checks tombstones presence via tombstone
+// source, caches it checks its expiration.
+//
+// Must be created via NewGetter function. `var` and
+// `Getter{}` declarations leads to undefined behaviour
+// and may lead to panics.
+type Getter struct {
+	cache *lru.Cache
+
+	log *zap.Logger
+
+	tsSource Source
+}
+
+// GetTombstone checks the tombstone presence in the system in the
+// following order:
+//  * 1. Local LRU cache;
+//  * 2. Tombstone source.
+//
+// If a tombstone was successfully fetched (regardless of its expiration)
+// it is cached in the LRU cache.
+func (g *Getter) GetTombstone(ctx context.Context, a *addressSDK.Address, epoch uint64) bool {
```

Comment:
Why is it called `GetTombstone`, but returns `bool`?

Reply:
one more awful naming. `TombstoneIsAvailable`, `IsAvailable`, `TobmstoneIsExpired`?

Reply:
`IsTombstoneAvailable`, `IsTombsoneAlive` look good to me.

Reply:
i have chosen `IsTombstoneAvailable`